Sources/App/CRP/SlackCommand+CRP.swift

```diff
@@ -70,30 +72,24 @@ extension SlackCommand {
 }
 
 
-/// Filters the CHANGELOG entries then orders and formats the CHANGELOG text
-///
-/// - Parameters:
-///   - commits: The list of commits gathered between last release and current one
-///   - release: The release for which to build the CHANGELOG text for
-/// - Returns: The text containing the filtered and formatted CHANGELOG, grouped and ordered by jira board
-private func formatChangelog(using commits: [String], for release: GitHubService.Release) -> String {
-    // Only keep SDK commits if release is for SDK
-    let filteredMessages = release.isSDK ? commits.filter { $0.contains("#SDK") || $0.contains("[SDK-") } : commits
+struct ChangelogSection {
+    let board: String?
+    let commits: [(message: String, ticket: JiraService.TicketID?)]
 
-    // Group then sort the changes by JIRA boards (unclassified last)
-    let grouped = Dictionary(grouping: filteredMessages) { JiraService.TicketID(from: $0)?.board }
-        .sorted { e1, e2 in e1.key ?? "ZZZ" < e2.key ?? "ZZZ" }
+    /// Filters the CHANGELOG entries then orders and formats the CHANGELOG text
+    ///
+    /// - Parameters:
+    ///   - commits: The list of commits gathered between last release and current one
+    ///   - release: The release for which to build the CHANGELOG text for
+    /// - Returns: The text containing the filtered and formatted CHANGELOG, grouped and ordered by jira board
+    static func makeSections(from commits: [String], for release: GitHubService.Release) -> [ChangelogSection] {
+        // Only keep SDK commits if release is for SDK
+        let filteredMessages = release.isSDK ? commits.filter { $0.contains("#SDK") || $0.contains("[SDK-") } : commits
```

Comment:
`$0.contains("[SDK-")`

Wouldn't be safer to match `SDK-` without the `[`? I think we have been following this convention but we may miss it by mistake and AFAIK we don't have any rule to enforce the [ ] around the tickets.

Reply:
good point. I didn't want to accidentally match some false positives (e.g. `"Fix SDK-related issue"`) though, but not sure it's worth it. Maybe I should instead regex-match against `SDK-[0-9]+` to be safer after all.